File: find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        players = set()
        loss_count = defaultdict(int)
        
        for winner, loser in matches:
            players.add(winner)
            players.add(loser)
            
            loss_count[loser] += 1
        
        zero_loss, one_loss = [], []
        for player in players:
            if player not in loss_count:
                zero_loss.append(player)
            elif loss_count[player] == 1:
                one_loss.append(player)
        
        return [sorted(zero_loss), sorted(one_loss)]

File: find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Set<Integer> players = new HashSet<>();
        Map<Integer, Integer> lossCount = new HashMap<>();
        List<List<Integer>> result = new ArrayList<>();
        
        for(int[] match : matches){
            int winner = match[0];
            int loser = match[1];
            players.add(winner);
            players.add(loser);
            
            lossCount.put(loser, lossCount.getOrDefault(loser, 0) + 1);
        }
        
        List<Integer> oneLoss = new ArrayList<>();
        List<Integer> noLoss = new ArrayList<>();
        for(int player : players){
            if(!lossCount.containsKey(player)){
                noLoss.add(player);
            }else if(lossCount.get(player) == 1){
                oneLoss.add(player);
            }
        }
        
        Collections.sort(noLoss);
        Collections.sort(oneLoss);
        
        result.add(noLoss);
        result.add(oneLoss);
        return result;
    }
}